### PR TITLE
feat(sdk-core): update the sdk core dependency

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
Add retry on 5xx errors

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,10 +50,10 @@
...
   ],
   "dependencies": {
-    "babel-runtime": "^6.3.19",
-    "contentful-sdk-core": "^2.3.0",
-    "json-stringify-safe": "^5.0.1",
-    "lodash": "^4.2.0"
+    "babel-runtime": "~6.3.19",
+    "contentful-sdk-core": "~2.5.0",
+    "json-stringify-safe": "~5.0.1",
+    "lodash": "~4.2.0"
```
Comment (johanneswuerbach) [inline package.json]: Out of curiosity, why did you change `^` to `~`?

Comment (Khaledgarbaya) [inline package.json]: turned out that some minor updates were breaking changes ¯\_(ツ)_/¯
